src/server/game/Grids/Notifiers/GridNotifiers.h

```diff
@@ -757,6 +757,29 @@ namespace Trinity
             NearestGameObjectEntryInObjectRangeCheck(NearestGameObjectEntryInObjectRangeCheck const&) = delete;
     };
 
+    class NearestReadyStateGameObjectEntryInObjectRangeCheck
+    {
+        public:
+            NearestReadyStateGameObjectEntryInObjectRangeCheck(WorldObject const& obj, uint32 entry, float range) : i_obj(obj), i_entry(entry), i_range(range) { }
+
+            bool operator()(GameObject* go)
+            {
+                if (go->GetEntry() == i_entry && i_obj.IsWithinDistInMap(go, i_range) && go->getLootState() == GO_READY)
```

Comment:
I've tried inheriting from more common check only adding concrete check here like so

```
if(Base::operator()(go) && concrete checks..)
```

Didn't work the way I thought it will, should I try it again or this also works?

Reply:
whats the purpose you were aiming for?

Reply:
@ccrs , well I thought that inheriting will remove a lot of boilerplate code and reduce duplications as well